### ISSUE TITLE
Fundamentally redesign parser

### ISSUE THREAD

**Issue by jonsmithers** [MergeMyPullRequest/order-splitter#30]
I think the input/parser design is fundamentally limiting. When the user copies the orderup receipt and pastes it into our textarea, it gets transformed from html into plaintext, so we lose lots of useful information. Consequently, we're make the user enter sales tax, processing fee, and delivery fee manually. 

We can just let the user paste html instead of plain text by using the [contenteditable attribute](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Editable_content) on on a `<div>`. Then we'd have the html to work with. (Also, we could re-use the same parser in a "live preview" in the chrome-extension.)

tl;dr The parser should be html-based, not text-based.

**Comment by dumbbillyhardy** [comment]
How would they get the html though? I think they would have to enter dev tools on orderup, which would be a bad experience I feel.

**Comment by jonsmithers** [comment]
They've already been copying html this whole time. We just need to accept html instead of text.

If you copy anything on this webpage and paste it into word document, it'll paste as rich text or something. If you paste it into order splitter's `<textarea>` element, it'll paste as text. If you paste into a `<div contenteditable="true">` element, it'll paste as html.

**Comment by jonsmithers** [comment]
Evernote.com uses content editable divs. See how I pasted html?

![screen shot 2017-06-21 at 3 48 51 pm](https://user-images.githubusercontent.com/2660313/27403650-2a35c49e-5699-11e7-8e0d-9421fd0c1a9e.png)

**Comment by dumbbillyhardy** [comment]
WOAH! That's awesome! I'm sold.
